LeaderboardBackend/Services/Impl/EmailSender.cs

```diff
@@ -0,0 +1,83 @@
+using MailKit.Net.Smtp;
+using Microsoft.Extensions.Options;
+using MimeKit;
+
+namespace LeaderboardBackend.Services;
+
+public class EmailSender : IEmailSender
+{
+    private readonly EmailSenderConfig _config;
+    private readonly ILogger<EmailSender> _logger;
+    private readonly ISmtpClient _smtpClient;
+    private readonly MailboxAddress _sender;
+    private readonly SemaphoreSlim _semaphore;
+
+    public EmailSender(IOptions<EmailSenderConfig> config, ILogger<EmailSender> logger, ISmtpClient smtpClient)
+    {
+        _config = config.Value;
+        _logger = logger;
+        _smtpClient = smtpClient;
+        _sender = new MailboxAddress(name: _config.SenderName, address: _config.SenderAddress);
+        _semaphore = new SemaphoreSlim(1);
+    }
+
+    public async Task SendEmailAsync(string recipientAddress, string subject, string htmlMessage)
```

Comment:
I doubt we will need this. Better add it when we actually need it.

The process that triggered the email and the queuing will be within the same DB transaction.

Reply:
Yes but we need to know if our email service is failing to send messages. That's a big deal. Say the email service encounters a problem and for whatever reason emails are not being sent. That means nobody will be able to register or recover their account. This method needs to report whether the email was successfully queued for sending so that we can potentially detect this error and investigate it.